src/rmc.rs

```diff
@@ -390,15 +263,19 @@ impl<'a, H: Signable + Hash + Eq + Clone + Debug, MK: MultiKeychain> ReliableMul
         for recipient in 0..self.node_count.0 {
             let recipient = NodeIndex(recipient);
             self.network_tx
-                .unbounded_send(OutgoingMessage {
-                    message: message.clone(),
-                    recipient,
-                })
-                .expect("Sending message shouldd succeed");
+                .unbounded_send((recipient, message.clone()))
+                .expect("Sending message should succeed");
+        }
+    }
+
+    pub fn get_multisigned(&self, hash: &H) -> Option<Multisigned<'a, H, MK>> {
+        match self.hash_states.get(hash)? {
+            HashState::SignatureAggregation { .. } => None,
+            HashState::Completed { complete } => Some(complete.clone()),
         }
     }
 
-    pub async fn _next_multisigned_hash(&mut self) -> Option<Multisigned<'a, H, MK>> {
+    pub async fn next_multisigned_hash(&mut self) -> Option<Multisigned<'a, H, MK>> {
```

Comment:
Uh, I just noticed nothing here is thread-safe. You have to be polling this to make RMC work, but at the same time you want to sometimes call at least `start_rmc` and perhaps `get_multisigned`. Both these require touching the map, which is not protected by a mutex or anything.

If you removed `get_multisigned` (as mentioned before I'm not completely sure it's necessary), then maybe the other case could be resolved by fiddling with the incoming messages channel (and using that in `start_rmc`), otherwise we need a mutex I believe. We might need it anyway, because without it `next_multisigned_hash` has to be `mut`?

Reply:
The problem is that also `multisigned_hashes_rx` would need to be protected by a mutex, and in the method `next_multisigned_hash` we would have to lock on it. I don't see yet how we can solve it...

Reply:
But do we intend to use an RMC instance in more than one thread?

Reply:
Good point, perhaps not.